### PR TITLE
data structures

### PR DESCRIPTION
**Pull request by mapsam**
This creates a `QueryData` and `TilesObject` struct's used to collect information to pass into the worker. Thanks to some help from @springmeyer we are able to do this in a no-copy way by storing both in unique_ptr's. RAII logic safely contained in the `TileObject` struct to keep things clean. 

cc @springmeyer @GretaCB

### CODE REVIEW

File: src/vtquery.cpp [hunk 1]
```diff
@@ -9,24 +9,95 @@
...
+        : Base(callback),
+          query_data_(std::move(query_data)) {}
 
     // The Execute() function is getting called when the worker starts to run.
     // - You only have access to member variables stored in this worker.
     // - You do not have access to Javascript v8 objects here.
     void Execute() override {
         try {
+            // Get the object from the unique_ptr
+            //QueryData const& data = *query_data_;
```
Comment (GretaCB) [inline src/vtquery.cpp:98]: @mapsam Is the `//` intentional here? Is the Worker executing from somewhere else?

**Comment by GretaCB** [comment]
Love these structs

File: src/vtquery.cpp (hunk 1 above)
Comment (mapsam) [reply src/vtquery.cpp:98]: We're not doing any actual work with `data` yet, so wasn't able to compile with an unused variable!

Comment (springmeyer) [reply src/vtquery.cpp:98]: wrt to `wasn't able to compile with an unused variable!` I've added `-Wno-error=unused-variable` to the recommended list, which disables just this one warning from becoming an error. No problem commenting/avoiding this warning of course - which is best practice -
 but I've found it a little annoying to have it be a hard error while developing. See https://github.com/mapbox/cpp/issues/37#issuecomment-336200744

Comment (GretaCB) [reply src/vtquery.cpp:98]: Sweet, thanks @mapsam . Key word --> `yet` :)

**Pull request merged**